docs/modules/ROOT/pages/references/architecture/upgrade_controller.adoc

```diff
@@ -222,6 +222,21 @@ openshift_upgrade_controller_cluster_version_info{channel="stable-4.14",cluster_
 openshift_upgrade_controller_cluster_version_overlay_timestamp_seconds{channel="stable-4.15",cluster_id="XXX",from="2022-12-04T14:00:00Z"} 1.6701624e+09
 ----
 
+=== The controller should be able to block certain time windows for upgrades (for example public holidays) [[block-upgrade-time-windows]]
+
+The controller should be able to block certain time windows for upgrades.
+This allows us to prevent upgrades during public holidays or other special events.
+
+A `UpgradeSuspensionWindow` object blocks upgrades for a specific time window.
+The objects it's matched against can be defined through a selector.
+
+Matching `UpgradeConfig` objects won't create `UpgradeJob` objects during the time window.
+
+Matching `UpgradeJob` objects won't start the upgrade during the time window.
+Skipped jobs will be marked as successful with reason skipped.
+Success and finish hooks will be executed as normal.
```

Comment:
How is it handled currently with start hooks for a noop upgradejob? I assume start and success/finish hooks are all executed, right?

Do we need to ensure that we don't have success/finish hooks that expect certain actions to be executed in a corresponding start hook? I'm thinking about a certain customer's scale-down then scale-up requirements.

Reply:
Ok, just something that we need to be aware of before we roll this out. It introduces a new corner case where only some of the hooks are executed, thereby potentially violating some expectations on the behavior that we had previously.

Reply:
Yes. I would not expect too many of those weird jobs though. Since they should™️ be idempotent anyways.